### PR TITLE
Add com.github.vikdevelop.googlemaps

### PR DESCRIPTION
**Pull request by vikdevelop**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
I know it will fail

bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by hfiguiere** [comment]
At the top of the build log:

`(flatpak-builder:3595796): flatpak-builder-WARNING **: 16:18:58.868: Unknown property sources for type BuilderManifest`

This is the source of your problems.

(Usually I make sure it buid locally before submitting)

**Comment by hfiguiere** [comment]
And in plain english `sources` is inside the module, not at the top level.

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by hfiguiere** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot. build com.github.vikdevelop.googlemaps

File: com.github.vikdevelop.googlemaps.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=wayland
+  - --socket=fallback-x11
+modules:
+  - name: googlemaps
+    buildsystem: simple
+    build-options:
+       append-path: /usr/lib/sdk/node14/bin
```
Comment (nedrichards) [inline com.github.vikdevelop.googlemaps.yml:19]: seems like a mismatch between the node16 extension and node14 here? (I haven't looked at this in detail, it may not be)

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github,vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

File: com.github.vikdevelop.googlemaps.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+base: org.electronjs.Electron2.BaseApp
+base-version: '21.08'
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=wayland
+  - --socket=fallback-x11
+
+build-options:
+  append-path: /usr/lib/sdk/node14/bin
```
Comment (hfiguiere) [inline com.github.vikdevelop.googlemaps.yml:17]:
```suggestion
  append-path: /usr/lib/sdk/node16/bin
```

Since you use the node16 SDK extension.

File: com.github.vikdevelop.googlemaps.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      dest-filename: run-a.sh
+      commands:
+        - zypak-wrapper.sh /app/main/googlemaps "$@"
+
+  - name: node_modules_google_maps
+    buildsystem: simple
+    build-options:
+       env:
+          XDG_CACHE_HOME: /run/build/googlemaps/flatpak-node/cache
+          npm_config_cache: /run/build/googlemaps/flatpak-node/npm-cache
```
Comment (hfiguiere) [inline com.github.vikdevelop.googlemaps.yml:50]: `/run/build/googlemaps` is gone at that point since the module is build.

File: com.github.vikdevelop.googlemaps.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+          npm_config_offline: 'true'
+    build-commands:
+    - npm install --offline
+    - cp -R node_modules /app/main/node_modules
+    sources:
+      - type: archive
+        url: https://github.com/vikdevelop/googlemaps/releases/download/1.0.1/node-modules_googlemaps.zip
```
Comment (hfiguiere) [inline com.github.vikdevelop.googlemaps.yml:57]: instead you should use the flatpak-builder-tools to generate a manifest for the dependencies.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
BTW you'd save a lot of time trying to build it locally first.

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps

**Comment by vikdevelop** [comment]
bot, build com.github.vikdevelop.googlemaps
